### PR TITLE
Cleanup build & add github actions

### PR DESCRIPTION
**Pull request by iloskutov**
Runs tests and clang-format check on commit and pull request

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install dependencies
+      run: sudo apt-get -y install python3-pip ninja-build clang-format
+    - name: Install latest Mesone
```
Comment (truvorskameikin) [inline .github/workflows/test.yml:18]: Mesone or Meson?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      run: pip3 install --user meson
+    - name: Prepare SDK
+      run: wget https://download.panic.com/playdate_sdk/Linux/PlaydateSDK-latest.tar.gz && mkdir ../PDSDK && tar xvf PlaydateSDK-latest.tar.gz --strip-components=1 -C ../PDSDK/
```
Comment (truvorskameikin) [inline .github/workflows/test.yml:21]: Why do we need Playdate SDK? All tests are now just pure C++ tests without any dependencies on Playdate.

File: Lib/game.cpp [hunk 3]
```diff
@@ -66,8 +66,7 @@ class Game {
 
     const char* error = 0;
     enemy_sprite_ = playdate_->graphics->loadBitmap("data/nazi.png", &error);
-    if (error)
-      playdate_->system->logToConsole("error: %s", error);
+    if (error) playdate_->system->logToConsole("error: %s", error);
```
Comment (truvorskameikin) [inline Lib/game.cpp:69]: Maybe add {} here?

File: .github/workflows/test.yml (hunk 2 above)
Comment (truvorskameikin) [reply .github/workflows/test.yml:21]: No, I was wrong, having PD SDK makes a lot of sense. Most likely we will test something that just includes "pd_api.h".

File: Lib/game.cpp (hunk 3 above)
Comment (iloskutov) [reply Lib/game.cpp:69]: We need to update clang-format config for it

Comment (truvorskameikin) [reply Lib/game.cpp:69]: I think that clang-format can't do it, I think that it doesn't add new tokens but rearranges the existing. I think that clang-tidy can do that.

**Review by truvorskameikin — APPROVED** [review]

**Pull request merged**